### DISCUSSION TITLE
Problem Running ProjectTracker with Xamarin.Android UI

### DISCUSSION THREAD

**Discussion by kobi-shanti** [MarimerLLC/csla#3327] [Questions] [answered]
Please excuse the seemingly basic question but I'm having issues trying to get the ProjectTracker sample to running using the Xamarin.Android UI.

I've setup VS to run multiple startup projects (App Server and Xamarin.Android) and to use the Conveyor extension by Keyoti to allow the android emulator to communicate with my App Server running locally (as described here: [https://github.com/MarimerLLC/cslaforum/issues/873]). I have confirmed that the emulator is successfully hitting the endpoint on the App Server by setting a breakpoint in PostAsync in DataPortalController. I have also confirmed that the App Server is successfully getting results out of my database by running GetAsync in DataPortalController.

The issue that I am having is that when PostAsync returns results to the emulator, I am getting an error:

"The input is not a valid Base-64 string as it contains a non-base 64 character..."

I also noticed that the relevant Fetch methods aren't being hit when PostAsync runs. In an attempt to figure the issue out, I tried replacing the contents of PostAsync, with a call to DataPortal.FetchAsync<Library.ProjectList>(). This successfully hit the relevant Fetch method, however the emulator doesn't seem to ever get the returned data or respond by updating the info in the view.

Am I missing some configuration step or something else obvious? The readme seems to suggest that all I need to do to get the sample to run is to setup the App Server so that it is available to the emulator and to update the app.xaml.cs with the location of my App Server.

Any help is greatly appreciated as I have been banging my head against the wall trying to figure this out.

Please let me know if any other info is required.

**Comment by rockfordlhotka** [comment]
Unfortunately ProjectTracker has fallen out of date, and one of my big projects over the next couple months is to update it for modern CSLA.

Right now you should look at the more platform targeted samples in the `Samples` directory.

**Reply by kobi-shanti** [reply]
Hey Rocky,

Thanks for coming back to me so quickly. We're looking to build a Xamarin app using an existing CSLA business object library and I am trying to find a template to work with for this project. One of the things that drew me to the project tracker sample was that it provided a full demo including the App Server and a more functional DAL. The Xamarin specific sample doesn't provide this, with no App Server and only a mock DB implemented. Is there any documentation or a sample available that is more up to date and demonstrates the server side of the CSLA implementation for Xamarin?

**Reply by rockfordlhotka** [reply]
I hear you, and updating ProjectTracker is something high on my to-do list.

I should say though, that the new ProjectTracker will have a MAUI client, not a Xamarin client. At this point I view Xamarin as legacy tech, superseded by MAUI.

You are right too, in that the only client-server sample right now (I think) is the BlazorExample, which has a smart client that interacts with an app server. The good news, is that the data portal client configuration is the same across all platforms, so the way the data portal is configured in the BlazorExample client app is the same as in MAUI or Xamarin. The server is likewise identical across all scenarios.

**Reply by rockfordlhotka** [reply]
fwiw, the primary reason the MauiExample app doesn't have an app server is that it can't be run in an Android emulator or iOS simulator against a localhost app server - at least not without a lot of extra work - so for demo purposes it is far simpler to skip that aspect to get a beginner going with the basics.

The MAUI _Windows_ client would work fine (afaik), because it can access localhost without issue.

**Reply by kobi-shanti** [reply]
We are looking to eventually migrate to MAUI once it is mature and stable enough, though we're not convinced of this just yet. It looks like promising tech so hopefully it isn't too far off.

I will check out the blazor sample and see if I can get the app server working with Xamarin.

Thanks again for your assistance and all your work with CSLA!

**Reply by kobi-shanti** [reply, marked as answer]
For anyone else running into similar issues trying to configure a n-tier CSLA Xamarin project, I managed to make some progress using the SimpleNTier sample as a base instead of ProjectTracker. I was able to get results back to the app by ensuring this line of code was excluded from the Xamarin client's CSLA configuration in App.xaml.cs (it was included in the ProjectTracker sample):

![image](https://github.com/MarimerLLC/csla/assets/130413276/5186fb74-dd11-4b0a-9db2-8912e2baf161)

For additional reference, we're looking at the CSLA 5.5.3 samples.
